UnityProject/Assets/Scripts/Gateway/StationGateway.cs

```diff
@@ -0,0 +1,178 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using Mirror;
+using UnityEngine;
+
+/// <summary>
+/// For the station Gate, only connects to a gate this script will pick
+/// </summary>
+public class StationGateway : NetworkBehaviour
+{
+	[SerializeField]
+	private SpriteRenderer SpriteBaseTop, SpriteBaseBottom, SpriteBaseCentre, SpriteBaseRightTop, SpriteBaseRightMiddle, SpriteBaseRightBottom, SpriteBaseLeftTop, SpriteBaseLeftMiddle, SpriteBaseLeftBottom;
+	//TODO animate centre
+
+	[SerializeField]
+	private Sprite[] Online;
+	[SerializeField]
+	private Sprite[] Offline;
+	[SerializeField]
+	private Sprite[] PowerOff;//TODO connect gateway to APC
+
+	[SerializeField]
+	private List<GameObject> Worlds = new List<GameObject>();//List of worlds available to be chosen
+
+	private GameObject SelectedWorld;// The world from the list that was chosen
+
+	private bool HasPower = true;// Not used atm
+
+	private bool IsConnected;
+
+	[SerializeField]
+	private int RandomCountBegining = 300; //Defaults to between 5 and 20 mins gate will open.
+	[SerializeField]
+	private int RandomCountEnd = 1200;
+
+	private RegisterTile registerTile;
+
+	private Matrix Matrix => registerTile.Matrix;
+
+	public string WorldName = "The Station";
+
+	private Vector3Int Position;
+
+	private string Message;
+
+	[Server]
+	private void Start()
+	{
+		registerTile = GetComponent<RegisterTile>();
+		Position = registerTile.WorldPosition;
+
+		SetOffline();
+
+		var count = Random.Range(RandomCountBegining, RandomCountEnd);
+		Invoke("WorldSetup", count);
+	}
+
+	[Server]
+	private void WorldSetup()//do message here as well
+	{
+		//Selects Random world
+		SelectedWorld = Worlds[Random.Range(0, Worlds.Count)];
+
+		if (SelectedWorld == null) return;
+
+		var selectedWorld = SelectedWorld.GetComponent<WorldGateway>();
+
+		Message = "Teleporting to: " + selectedWorld.WorldName;
+
+		if (selectedWorld.IsOnlineAtStart == false)
+		{
+			selectedWorld.IsOnlineAtStart = true;
+			selectedWorld.gameObject.SetActive(true);
+			selectedWorld.SetUp();
+		}
+
+		if (HasPower == true)
+		{
+			SetOnline();
+
+			var text = "Alert! New Gateway connection formed.\n\n Connection established to: " + SelectedWorld.GetComponent<WorldGateway>().WorldName;
+			CentComm.MakeAnnouncement(CentComm.CentCommAnnounceTemplate, text, CentComm.UpdateSound.alert);
+
+			loop();
+		}
+	}
+
+	[Server]
+	private void loop()
```

Comment:
This should be uppercase and more descriptive.

Reply:
done